addons/website/models/website.py

```diff
@@ -1340,3 +1341,235 @@ def _get_cached_values(self):
 
     def _get_cached(self, field):
         return self._get_cached_values()[field]
+
+    def _search_get_details(self, search_type, order, options):
+        """
+        Returns indications on how to perform the searches
+
+        :param search_type: type of search
+        :param order: order in which the results are to be returned
+        :param options: search options
+
+        :return list of search details obtained from the `website.searchable.mixin`'s `_search_get_detail()`
+        """
+        result = []
+        if search_type in ['pages', 'all']:
+            result.append(self.env['website.page']._search_get_detail(self, order, options))
+        return result
+
+    def _search_with_fuzzy(self, search_type, search, limit, order, options):
+        """
+        Performs a search with a search text or with a resembling word
+
+        :param search_type: indicates what to search within, 'all' matches all available types
+        :param search: text against which to match results
+        :param limit: maximum number of results per model type involved in the result
+        :param order: order on which to sort results within a model type
+        :param options: search options from the submitted form containing:
+            - allowFuzzy: boolean indicating whether the fuzzy matching must be done
+            - other options used by `_search_get_details()`
+
+        :return tuple containing:
+            - count: total number of results across all involved models
+            - results: list of results per model (see _search_exact)
+            - fuzzy_term: similar word against which results were obtained, indicates there were
+                no results for the initially requested search
+        """
+        fuzzy_term = False
+        search_details = self._search_get_details(search_type, order, options)
+        if search and options.get('allowFuzzy', True):
+            fuzzy_term = self._search_find_fuzzy_term(search_details, search)
+            if fuzzy_term:
+                count, results = self._search_exact(search_details, fuzzy_term, limit, order)
+                if fuzzy_term == search:
+                    fuzzy_term = False
+            else:
+                count, results = 0, search_details
+        else:
+            count, results = self._search_exact(search_details, search, limit, order)
+        return count, results, fuzzy_term
+
+    def _search_exact(self, search_details, search, limit, order):
+        """
+        Performs a search with a search text
+
+        :param search_details: see `_search_get_details()`
```

Comment:
```suggestion
        :param search_details: see :meth:`~._search_get_details`
```
same in other pseudo-references. Incidentally docstrings are reStructuredText so literal text is double-backticks, not single.